source/PlayerInfo.cpp

```diff
@@ -3672,6 +3672,14 @@ void PlayerInfo::RegisterDerivedConditions()
 	pluginProvider.SetHasFunction(pluginFun);
 	pluginProvider.SetGetFunction(pluginFun);
 
+	auto &&destroyedPersonProvider = conditions.GetProviderPrefixed("destroyed person: ");
+	auto destroyedPersonFun = [](const string &name) -> bool
+	{
+		return GameData::Persons().Has(name.substr(strlen("destroyed person: ")));
```

Comment:
This looks like it's just checking if a person with the given name exists in the collection, not if that person has been destroyed?
We have:
https://github.com/endless-sky/endless-sky/blob/3860eca102a93b77df41c94658a64ea6d85bc7a1/source/PlayerInfo.cpp#L4321-L4324
Which looks very different to what you're doing here.

Reply:
Woops ig I assumed they were destroyed to be efficient